### PR TITLE
Issue 8: Re-enable json row decoder

### LINKED ISSUE #8: Commented `decodeTree` in JsonRowDecoder
In "src/main/java/com/facebook/presto/pravega/decoder/JsonRowDecoder.java", the call to: 

```      
      ((TypedRecordValue) record).setDecodedValue(jsonRowDecoder.decodeTree(event.asJson())); 
```
 
 is commented out and needs to be uncommented. We have added the `decodeTree` method to `JsonRowDecoder`, and that source code is no longer in the tree, since we are now referencing it by jars.

### PR DESCRIPTION
**Pull request by adrdc**
re-enable Json decoding.  previously json decoding had changed files in common prestodb libs which we are not modifyig now.

Fixes #8

### CODE REVIEW

File: src/main/java/com/facebook/presto/pravega/decoder/JsonRowDecoderFactory.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.Set;
+
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+
+public class JsonRowDecoderFactory
```
Comment (fpj) [inline src/main/java/com/facebook/presto/pravega/decoder/JsonRowDecoderFactory.java:42]: Can we unit test this class?

File: NOTICE [hunk 2]
```diff
@@ -1 +1,5 @@
 Copyright (c) 2021 Pravega Authors.
+
+This software contains source code from Presto, distributed under
```
Comment (fpj) [inline NOTICE:3]: I don't understand which code we are bringing from Presto here, is it the JsonRowDecoderFactory?

**Review by fpj — CHANGES_REQUESTED** [review]

File: NOTICE (hunk 2 above)
Comment (adrdc) [reply NOTICE:3]: JsonRowDecoderFactory, and half of JsonRowDecoder.  had been looking for guidance on what to do w.r.t class headers and NOTICE file for these.

**Review by karansinghneu — APPROVED** [review]

File: src/main/java/io/pravega/connectors/presto/decoder/JsonRowDecoderFactory.java [hunk 3]
```diff
@@ -12,8 +10,11 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ *
+ * Note: This class file is from PrestoDb
```
Comment (fpj) [inline src/main/java/io/pravega/connectors/presto/decoder/JsonRowDecoderFactory.java:14]: Minor: can you add the commit id in PrestoDB from which you extracted this code?

File: src/test/java/io/pravega/connectors/presto/decoder/TestJsonRowDecoder.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Test
+public class TestJsonRowDecoder
```
Comment (fpj) [inline src/test/java/io/pravega/connectors/presto/decoder/TestJsonRowDecoder.java:43]: Minor: Let's follow the pattern of adding test to the end of the class name,  `JsonRowDecoderTest` instead in this case.

**Review by chipmaurer — APPROVED** [review]

File: src/main/java/io/pravega/connectors/presto/decoder/JsonRowDecoderFactory.java (hunk 3 above)
Comment (adrdc) [reply src/main/java/io/pravega/connectors/presto/decoder/JsonRowDecoderFactory.java]: added

File: src/test/java/io/pravega/connectors/presto/decoder/TestJsonRowDecoder.java (hunk 4 above)
Comment (adrdc) [reply src/test/java/io/pravega/connectors/presto/decoder/TestJsonRowDecoder.java]: renamed

**Review by fpj — APPROVED** [review]

**Pull request merged**